TypeCobol/Compiler/Diagnostics/CodeElementCheckers.cs

```diff
@@ -201,52 +201,58 @@ public static void OnCodeElement(CallStatement statement, CodeElementsParser.Cal
 
                 //SpecialRegister if LENGTH OF, LINAGE-COUNTER, ...
-                var specialRegister = input.StorageAreaOrValue != null
-                    ? input.StorageAreaOrValue.StorageArea as StorageAreaPropertySpecialRegister
+                var specialRegister = inputParameter.StorageAreaOrValue != null
+                    ? inputParameter.StorageAreaOrValue.StorageArea as StorageAreaPropertySpecialRegister
                     : null;
 
                 if (isFunctionCallResult)
-                    DiagnosticUtils.AddError(statement, "CALL .. USING: Illegal function identifier", context);
+                    DiagnosticUtils.AddError(statement, "CALL .. USING: Illegal function identifier", errorPosition);
 
                 if (specialRegister != null && specialRegister.SpecialRegisterName.TokenType == TokenType.LINAGE_COUNTER)
-                    DiagnosticUtils.AddError(statement, "CALL .. USING: Illegal LINAGE-COUNTER", context);
+                    DiagnosticUtils.AddError(statement, "CALL .. USING: Illegal LINAGE-COUNTER", errorPosition);
 
 
-                if (input.SharingMode != null)
+                if (inputParameter.SharingMode != null)
                 {
                     //BY REFERENCE
-                    if (input.SharingMode.Value == ParameterSharingMode.ByReference)
+                    if (inputParameter.SharingMode.Value == ParameterSharingMode.ByReference)
                     {
                         if (specialRegister != null && specialRegister.SpecialRegisterName.TokenType == TokenType.LENGTH)
                             DiagnosticUtils.AddError(statement,
-                                "CALL .. USING: Illegal LENGTH OF in BY REFERENCE phrase", context);
+                                "CALL .. USING: Illegal LENGTH OF in BY REFERENCE phrase", errorPosition);
 
-                        if (input.StorageAreaOrValue != null && input.StorageAreaOrValue.IsLiteral)
+                        if (inputParameter.StorageAreaOrValue != null && inputParameter.StorageAreaOrValue.IsLiteral)
                             DiagnosticUtils.AddError(statement,
-                                "CALL .. USING: Illegal <literal> in BY REFERENCE phrase", context);
+                                "CALL .. USING: Illegal <literal> in BY REFERENCE phrase", errorPosition);
                     }
 
                     //BY VALUE
-                    if (input.IsOmitted && input.SharingMode.Value == ParameterSharingMode.ByValue)
-                        DiagnosticUtils.AddError(statement, "CALL .. USING: Illegal OMITTED in BY VALUE phrase", context);
+                    if (inputParameter.IsOmitted && inputParameter.SharingMode.Value == ParameterSharingMode.ByValue)
+                        DiagnosticUtils.AddError(statement, "CALL .. USING: Illegal OMITTED in BY VALUE phrase", inputParameter.Omitted?.Token);
```

Comment:
Omitted is not null

Reply:
Nullity test has been removed and a comment has been added to emphasize the difference of the position for the diagnostic.